omniNotes/src/androidTest/java/it/feio/android/omninotes/utils/SortNotes.java

```diff
@@ -0,0 +1,74 @@
+package it.feio.android.omninotes.utils;
+
+import android.widget.ListView;
+
+import androidx.test.espresso.ViewInteraction;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import it.feio.android.omninotes.R;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isChecked;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withParent;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static it.feio.android.omninotes.utils.EspressoUtils.checkIfMatches;
+import static it.feio.android.omninotes.utils.EspressoUtils.childAtPosition;
+import static it.feio.android.omninotes.utils.EspressoUtils.clickOnButton;
+import static it.feio.android.omninotes.utils.EspressoUtils.getText;
+import static it.feio.android.omninotes.utils.NewNoteStringVariables.titleButton;
+import static org.hamcrest.Matchers.allOf;
+
+public class SortNotes {
+    /**
+     * This method is sorting the list of notes and stores the titles in an array and return the array with the titles
+     *
+     * @param arrayWithTitles - in this parameter are stored the titles after sorting by title
+     * @param numberOfTimes   - this is the parameter of how many notes were added
+     * @return -  the array with titles
+     */
+    public static List<String> getArrayWithTitlesAfterSort(List<String> arrayWithTitles, int numberOfTimes) {
+        int index = 0;
+
+        sortBy(titleButton);
+        while (index < numberOfTimes) {
+            arrayWithTitles.add(getText(allOf(withParent(childAtPosition(childAtPosition(childAtPosition(withId(R.id.list), index), 0), 1)), withId(R.id.note_title))));
+            index++;
+        }
+        return arrayWithTitles;
+    }
+
+    /**
+     * This method sorts the list with notes by a text
+     *
+     * @param buttonText - the sorting is done by the text selected e.g. Title, Creation date
+     */
+    public static void sortBy(String buttonText) {
+        Map<String, Integer> mapWithSortingOptions = new HashMap<>();
+        mapWithSortingOptions.put("Title", 0);
+        mapWithSortingOptions.put("Creation date",1);
+        mapWithSortingOptions.put("Last modification date",2);
+        mapWithSortingOptions.put("Reminder date",3);
+
+
+        ViewInteraction sortButton = onView(withId(R.id.menu_sort));
+        clickOnButton(sortButton);
+        do {
+            clickOnButton(onView(withText(buttonText)));
+            clickOnButton(sortButton);
+
+        } while (!isSelected(mapWithSortingOptions.get(buttonText)));
```

Comment:
one more thing: this loop can (and will sometimes) loop forever. you need a stop condition. something like a max number of tries. between tries, try to wait 1 second or so to allow the app to process your request.

Reply:
one workaround to avoid repetitive checks and selections is to put the focus on other element after the first button selection before doing again the check, for example: after you selected once creation date or title button, click on some other button (the three-dots button near the sort button) and then redo the check

Reply:
> one more thing: this loop can (and will sometimes) loop forever. you need a stop condition. something like a max number of tries. between tries, try to wait 1 second or so to allow the app to process your request.

added a 10 max number of tries and a 2 sec wait between them

Reply:
> one workaround to avoid repetitive checks and selections is to put the focus on other element after the first button selection before doing again the check, for example: after you selected once creation date or title button, click on some other button (the three-dots button near the sort button) and then redo the check

added a clicking on the add new notes button (twice - once for open and once for closing)